### DISCUSSION TITLE
Issue with CSS comments on GrapesJS Web Builder

### DISCUSSION THREAD

**Discussion by Ashwary-Jharbade** [GrapesJS/grapesjs#5721] [General]
Is there any way to support and display CSS comments on GrapeJS CSS style viewer ?

When I tried adding some CSS styles using "Custom Code" block with some comments, GrapesJS removed the commented styles and replaced some of the style properties with "undefined: undefined".

I have taken some screenshot for our discussion reference.

1. I have an element with class "mybox"
![Grapesjs_box](https://github.com/GrapesJS/grapesjs/assets/55127977/99262301-823b-4fdf-9400-4c171f5550a0)

2. I have used "Custom Code" block element to add some styles and comments.
![GrapesJS_comments](https://github.com/GrapesJS/grapesjs/assets/55127977/bc972315-720f-43ab-8f20-ab349103941e)

3. Result: Once saved and tried to view my styles on CSS viewer GrapesJS removed a entire block of CSS and replaced some CSS styles with undefined values.
![Grapesjs_result](https://github.com/GrapesJS/grapesjs/assets/55127977/e440382b-0c77-470a-a56b-4641bb296504)

Note: I have experienced this on web version of GrapesJS Web Builder
Web Builder Url - https://grapesjs.com/demo.html

**Comment by mosh-tudor** [comment]
> and display CSS comments on GrapeJS CSS style viewer 

But how do you imagine this?

The editor cannot know which part of the code the comment belongs to, and after a few clicks in StyleManager, your comment won't be where you expect it to be.

I think you should remove all css/html comments before saving.

**Reply by Ashwary-Jharbade** [reply]
Hi @mosh-tudor ,
Good to know you are also curious about it. But seems like CSS does not get impacted much with repositioning of block elements on Web Builder. Mostly CSS gets appended. And GrapesJS already supports HTML comments.
